Feature/edit profile

Vladyslav-Shvorobuk opened the pull request.

Commit:
feature/edit-profile
change function onFileChanged

Commit:
feature/edit-profile
fixed merge conflict

motifej commented on src/app/pages/edit-profile/edit-profile.controller.js:
```diff
@@ -36,8 +35,7 @@ function EditProfileController($log, $scope, $rootScope, userProfileService, syn
       $scope.formInfo.ava = reader.result;
-      ava.src = reader.result;
-
+      $scope.$apply();
```
don't ever use  _$scope.$apply()_ !!!

Vladyslav-Shvorobuk replied:
I changed it to $scope.$applyAsync().

motifej commented on src/app/pages/edit-profile/edit-profile.controller.js:
```diff
@@ -18,6 +18,7 @@ function EditProfileController($log, $scope, $rootScope, userProfileService, syn
       }
+
       $scope.success = true;
       setTimeout(function () {
```
don't ever use any native methods !!!

Vladyslav-Shvorobuk replied:
I changed it to $timeout().

Commit:
feature/edit-profile,
1) change setTimeout to $timeout
2) change apply to applyAsync

motifej commented on src/app/pages/edit-profile/edit-profile.controller.js:
```diff
@@ -18,26 +18,25 @@ function EditProfileController($log, $scope, $rootScope, userProfileService, syn
-      setTimeout(function () {
+      $timeout(function () {
         $state.go('profile');
       }, 1500);
```
why do you decide use this time for waiting?

Vladyslav-Shvorobuk replied:
I based on my own experience, I tried to use another time, but this time seemed to me optimal.

Vladyslav-Shvorobuk replied:
and this time will be enought for reading message about successful saving.

Vladyslav-Shvorobuk replied:
I delete this $timeout.

Commit:
feature/edit-profile,
1) rewrite profile, edit-profile to classes
2) delete alert-message success and styles for it
3) delete $timeout

Commit:
feature/edit-profile,
add directive on-file-change

Commit:
Merge branch 'development' of github.com:Arkant/AngularJSMiniApp into feature/edit-profile

Commit:
feature/edit-profile,
rewrite function createFormInfo and saveToCurrentUser using destracture

Commit:
feature/edit-profile,
1) delete ng-hide, add ng-message to edit-profile.html
2) make more readable code in edit-profile.html

Commit:
feature/edit-profile,
1) delete ng-class
2) relocate profile from scope to this

Commit:
Merge branch 'development' of github.com:Arkant/AngularJSMiniApp into feature/edit-profile

Commit:
feature/edit-profile,
add default ava

Commit:
feature/edit-profile,
delete unnecessary ckeck

Commit:
feature/edit-profile,
delete unnecessary link

motifej reviewed (approved):
good Job!
